cmd/clusterctl/client/repository/components.go

```diff
@@ -194,18 +189,23 @@ type ComponentsOptions struct {
 // 4. Ensure all the ClusterRoleBinding which are referencing namespaced objects have the name prefixed with the namespace name
 // 5. Set the watching namespace for the provider controller
 // 6. Adds labels to all the components in order to allow easy identification of the provider objects
-func NewComponents(provider config.Provider, configClient config.Client, rawyaml []byte, options ComponentsOptions) (*components, error) {
-	// Inspect the yaml read from the repository for variables.
-	variables := inspectVariables(rawyaml)
+func NewComponents(provider config.Provider, configClient config.Client, processor yaml.Processor, rawyaml []byte, options ComponentsOptions) (*components, error) {
 
-	// Replace variables with corresponding values read from the config
-	yaml, err := replaceVariables(rawyaml, variables, configClient.Variables(), options.SkipVariables)
+	variables, err := processor.GetVariables(rawyaml)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to perform variable substitution")
+		return nil, err
+	}
+
+	processedYaml := rawyaml
+	if !options.SkipVariables {
```

Comment:
I'm not sure about this test.
SkipVariables imply to process the template, but to not fail in case some variable is missing; instead, if I got this right this is changing the semantic of the flag by skipping template processing in this case

Reply:
It basically states that if you don't need variable substitution just return the original components yaml. But if variable substitution is required then process the yaml with variables.
For the existing Simple processor the processing is variable substitution that's why it seems like the `SkipVariables` option is used to gate the processing.
In the future if there are other types of processing for the infrastructure components we can refactor this then.

Reply:
Ok, I don't want to block on this and so l think that we can drop the partial variable substitution. 
Hopefully, this won't hurt users, and we land in a cleaner state.